src/new-client/src/app/lib/services/viewAvailabilityService.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  AvailabilityBlock,
+  AvailabilityResponse,
+  Booking,
+  clinicalServices,
+  FetchBookingsRequest,
+  Week,
+} from '@types';
+import dayjs from 'dayjs';
+import { fetchBookings } from './appointmentsService';
+
+const BuildWeeks = (year: number, month: number): Week[] => {
+  const weeks: Week[] = [];
+
+  const [lastDateMonth, lastDateYear] =
+    month === 11 ? [0, year + 1] : [month + 1, year];
+  const firstDate = new Date(year, month, 1);
+  const lastDate = new Date(lastDateYear, lastDateMonth, 0);
+  const numDays = lastDate.getDate();
+  const dayOfWeekCounter = firstDate.getDay();
+
+  let start = 1;
+  let end = 7;
+  if (dayOfWeekCounter === 0) {
+    end = 1;
+  } else {
+    end = 7 - dayOfWeekCounter + 1;
+  }
+
+  while (start <= numDays) {
+    weeks.push({
+      start: start,
+      end: end,
+      startMonth: month,
+      endMonth: month,
+      startYear: year,
+      endYear: lastDateYear,
+      bookedAppointments: [],
+    });
+    start = end + 1;
+    end = end + 7;
+    end = start === 1 && end === 8 ? 1 : end;
+    if (end > numDays) {
+      end = numDays;
+    }
+  }
+
+  return weeks;
+};
+
+export const getWeeksInMonth = (year: number, month: number): Week[] => {
+  const weeks = BuildWeeks(year, month);
+
+  if (weeks[0].start === 1) {
+    const beforeWeeks =
+      month === 0 ? BuildWeeks(year, 11) : BuildWeeks(year, month - 1);
+    weeks[0].start = beforeWeeks[beforeWeeks.length - 1].start;
+    weeks[0].startMonth = beforeWeeks[beforeWeeks.length - 1].startMonth;
+  }
+
+  if (weeks[weeks.length - 1].end === new Date(year, month + 1, 0).getDate()) {
+    const afterWeeks =
+      month === 11 ? BuildWeeks(year + 1, 0) : BuildWeeks(year, month + 1);
+    weeks[weeks.length - 1].end = afterWeeks[0].start;
+    weeks[weeks.length - 1].endMonth = afterWeeks[0].endMonth;
+  }
+
+  return weeks;
+};
+
+const getUnbookedCount = (
+  availability: AvailabilityResponse[],
+  week: Week,
+): number => {
+  let unbookedCount = 0;
+  if (!availability || availability.length === 0) {
+    return unbookedCount;
+  }
+
+  for (let a = 0; a < availability.length; a++) {
+    if (availability[a].availability.length === 0) {
+      continue;
+    }
+
+    const fromDate = weekStart(week);
+    const toDate = weekEnd(week);
+
+    const blocks: AvailabilityBlock[] = [];
+    availability[a].availability.filter(item => {
+      const date = new Date(item.date);
+      if (
+        date.getTime() >= fromDate.getTime() &&
+        date.getTime() <= toDate.getTime()
+      ) {
+        blocks.push(...item.blocks);
+      }
+    });
+
+    const countArray = blocks.map(b => b.count);
+    unbookedCount =
+      countArray.length === 0 ? 0 : countArray.reduce((sum, num) => sum + num);
+  }
+
+  return unbookedCount;
+};
+
+const weekStart = (week: Week): Date => {
+  return new Date(week.startYear, week.startMonth, week.start, 0, 0, 0);
```

Comment:
We might be able to save ourselves a lot of manual maths if we switch to DayJs, e.g. the [isoWeek plugin](https://day.js.org/docs/en/plugin/iso-week) would let us just go `date.startOf('isoWeek')`. Same goes for start of month/year etc. Might we be able to simplify the new `Week` type a little too, as we might not need the `monthStart`/`yearStart` properties for example?